File: leecode/401-500/448-找到所有数组中消失的数字.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        int n = nums.size();
        for (int i = 0; i < n; ++i) {
            int ind = abs(nums[i]) - 1;
            nums[ind] = -abs(nums[ind]);
        }
        vector<int> ans;
        for (int i = 0; i < n; ++i)
            if (nums[i] > 0)
                ans.emplace_back(i+1);
        return ans;
    }
};

File: leecode/1-500/401-500/448-找到所有数组中消失的数字.py
class Solution:
    def findDisappearedNumbers(self, nums):
        n = len(nums)
        for i in range(n):
            a = abs(nums[i]) - 1
            nums[a] = -abs(nums[a])
        res = []
        for i in range(n):
            if nums[i] > 0:
                res.append(i+1)
        return res
